samples/quickstart.js

```diff
@@ -14,60 +14,62 @@
  */
 
 'use strict';
+async function main() {
+  // [START dataproc_quickstart]
+  if (
+    !process.env.GCLOUD_PROJECT ||
+    !process.env.GOOGLE_APPLICATION_CREDENTIALS
+  ) {
+    throw new Error(
+      'Usage: GCLOUD_PROJECT=<project_id> GOOGLE_APPLICATION_CREDENTIALS=<path to json key> node #{$0}'
+    );
+  }
 
-// [START dataproc_quickstart]
-if (
-  !process.env.GCLOUD_PROJECT ||
-  !process.env.GOOGLE_APPLICATION_CREDENTIALS
-) {
-  throw new Error(
-    'Usage: GCLOUD_PROJECT=<project_id> GOOGLE_APPLICATION_CREDENTIALS=<path to json key> node #{$0}'
-  );
-}
-
-const dataproc = require('@google-cloud/dataproc');
+  const dataproc = require('@google-cloud/dataproc');
 
-const client = new dataproc.v1.ClusterControllerClient({
-  // optional auth parameters.
-});
+  const client = new dataproc.v1.ClusterControllerClient({
+    // optional auth parameters.
+  });
 
-const projectId = process.env.GCLOUD_PROJECT;
+  const projectId = process.env.GCLOUD_PROJECT;
 
-// Iterate over all elements.
-const region = 'global';
-const request = {
-  projectId: projectId,
-  region: region,
-};
+  // Iterate over all elements.
+  const region = 'global';
+  const request = {
+    projectId: projectId,
+    region: region,
+  };
 
-client.listClusters(request).then(responses => {
-  const resources = responses[0];
+  const [resources] = await client.listClusters(request);
   console.log('Total resources:', resources.length);
   for (let i = 0; i < resources.length; i += 1) {
     console.log(resources[i]);
   }
-});
 
-// Or obtain the paged response.
-const options = {autoPaginate: false};
-const callback = responses => {
```

Comment:
Mixing callback and promise based code is weird.  Please just run whatever callback code was supposed to be run linearly after the promise resolves.